admiral/pkg/clusters/admiralDatabaseClient.go

```diff
@@ -181,13 +201,58 @@ func ReadAndUpdateSyncAdmiralConfig(dbClient AdmiralDatabaseManager) error {
 	if IsDynamicConfigChanged(configData) {
 		log.Infof(fmt.Sprintf("task=%s, updating DynamicConfigData with Admiral config", common.DynamicConfigUpdate))
 		UpdateSyncAdmiralConfig(configData)
+
+		ctx := context.Context(context.Background())
+		//Process NLB Cluster
+		processLBMigration(ctx, rr, common.GetAdmiralParams().NLBEnabledClusters, &rr.AdmiralCache.NLBEnabledCluster, common.GetAdmiralParams().NLBIngressLabel)
+		//Process CLB Cluster
+		processLBMigration(ctx, rr, common.GetAdmiralParams().CLBEnabledClusters, &rr.AdmiralCache.CLBEnabledCluster, common.GetAdmiralParams().LabelSet.GatewayApp)
 	} else {
 		log.Infof(fmt.Sprintf("task=%s, no need to update DynamicConfigData", common.DynamicConfigUpdate))
 	}
 
 	return nil
 }
 
+func processLBMigration(ctx context.Context, rr *RemoteRegistry, updatedLBs []string, existingCache *[]string, lbLabel string) {
+
+	log.Infof("task=%s, Processing LB migration for %s. UpdateReceived=%s, ExistingCache=%s, ", common.LBUpdateProcessor, lbLabel, updatedLBs, existingCache)
+
+	for _, cluster := range getLBToProcess(updatedLBs, existingCache) {
+		for _, fetchService := range rr.remoteControllers[cluster].ServiceController.Cache.Get(common.NamespaceIstioSystem) {
+			if fetchService.Labels[common.App] == lbLabel {
+				log.Infof("task=%s, Processing LB migration for Cluster : %s", common.LBUpdateProcessor, cluster)
+				go handleServiceEventForDeployment(ctx, fetchService, rr, cluster, rr.GetRemoteController(cluster).DeploymentController, rr.GetRemoteController(cluster).ServiceController, HandleEventForDeployment)
+				go handleServiceEventForRollout(ctx, fetchService, rr, cluster, rr.GetRemoteController(cluster).RolloutController, rr.GetRemoteController(cluster).ServiceController, HandleEventForRollout)
+			}
+		}
+	}
+}
+
+func getLBToProcess(updatedLB []string, cache *[]string) []string {
+	var clusersToProcess []string
+	if cache == nil || len(*cache) == 0 {
+		*cache = updatedLB
+		clusersToProcess = updatedLB
+		return clusersToProcess
+	} else {
+		//Validate if New ClusterAdded
+		for _, clusterFromAdmiralParam := range updatedLB {
+			if !slices.Contains(*cache, clusterFromAdmiralParam) {
+				clusersToProcess = append(clusersToProcess, clusterFromAdmiralParam)
```

Comment:
we should add the new cluster in the cache, or is it being added somewhere else?